pkgs/tools/security/wordlists/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, runtimeShell
+, stdenv
+, tree
+, pkgs
+}:
+
+let
+  _wordlistData =
+    [ { name = "dirbuster"; expressionPath = ./dirbuster.nix; }
+      { name = "nmap"; expressionPath = ./nmap.nix; needsPackage = true; }
+      { name = "rockyou"; expressionPath = ./rockyou.nix; }
+      { name = "seclists"; expressionPath = ./seclists.nix; }
+      { name = "wfuzz"; expressionPath = ./wfuzz.nix; needsPackage = true; }
+    ];
+
+  scopedWordlists = lib.makeScope pkgs.newScope (self: with self;
+    let _packageWithLinkCommandFrom = acc: datum:
+      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
+      let parameters = {}
+        // lib.optionalAttrs ((builtins.elem "needsPackage" (builtins.attrNames datum)) && datum.needsPackage) { ${datum.name} = pkgs.${datum.name}; };
+      in
+        let p = callPackage datum.expressionPath parameters;
+        in acc // { ${datum.name} = p; };
+    in builtins.foldl' _packageWithLinkCommandFrom {} _wordlistData
+  );
+in let
+  # Given an array of wordlist packages,
+  # return an array of bash commands to create symlinks to the files shared by each package.
+  allLinksCommandFor = wordlistPackages:
+    lib.concatStringsSep "\n" (builtins.map (x: "ln -s ${x}/share/${x.passthru.shared} $out/share/${x.passthru.shared}") wordlistPackages);
+
+in let
+  packageConstructor = { withLists ? null, wordlistPackages ? [] }:
```

Comment:
This handling of `withLists` irritates me a bit. Why is it included passively (via passthru) in the derivation?

Reply:
> Why is it included passively (via passthru) in the derivation?

Because I didn't (and still don't) know how to have `security-wordlists` to both have `.pkgs.<...>` and `.withLists`.

There are two usages for this package:
  * one as a package to use, via `wordlists` and `wordlists_path` shell commands
  * one to allow a specific wordlist to be used a requirement in a derivation: e.g. `buildInputs = [ security-wordlists.pkgs.nmap ]`

> Why not simply returning a function, which takes withLists as input.

I am not too certain I understand what you propose, but it seems that having a function to take `withLists` as input would break the second case.